xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/PVR.h

```diff
@@ -0,0 +1,7939 @@
+   *
+   * Structure used to interface in "C" between Kodi and Addon.
+   *
+   * See @ref kodi::addon::PVRCapabilities for description of values.
+   */
+  typedef struct PVR_ADDON_CAPABILITIES
+  {
+    bool bSupportsEPG;
+    bool bSupportsEPGEdl;
+    bool bSupportsTV;
+    bool bSupportsRadio;
+    bool bSupportsRecordings;
+    bool bSupportsRecordingsUndelete;
+    bool bSupportsTimers;
+    bool bSupportsChannelGroups;
+    bool bSupportsChannelScan;
+    bool bSupportsChannelSettings;
+    bool bHandlesInputStream;
+    bool bHandlesDemuxing;
+    bool bSupportsRecordingPlayCount;
+    bool bSupportsLastPlayedPosition;
+    bool bSupportsRecordingEdl;
+    bool bSupportsRecordingsRename;
+    bool bSupportsRecordingsLifetimeChange;
+    bool bSupportsDescrambleInfo;
+    bool bSupportsAsyncEPGTransfer;
+    bool bSupportsRecordingSize;
+
+    unsigned int iRecordingsLifetimesSize;
+    struct PVR_ATTRIBUTE_INT_VALUE recordingsLifetimeValues[PVR_ADDON_ATTRIBUTE_VALUES_ARRAY_SIZE];
+  } PVR_ADDON_CAPABILITIES;
+
+  //}}}
+  //____________________________________________________________________________
+  //--==----==----==----==----==----==----==----==----==----==----==----==----==
+  //¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+  // "C" Definitions group 2 - PVR channel
+  //{{{
+
+  /*!
+   * @brief "C" PVR add-on channel.
+   *
+   * Structure used to interface in "C" between Kodi and Addon.
+   *
+   * See @ref kodi::addon::PVRChannel for description of values.
+   */
+  typedef struct PVR_CHANNEL
+  {
+    unsigned int iUniqueId;
+    bool bIsRadio;
+    unsigned int iChannelNumber;
+    unsigned int iSubChannelNumber;
+    char strChannelName[PVR_ADDON_NAME_STRING_LENGTH];
+    char strMimeType[PVR_ADDON_INPUT_FORMAT_STRING_LENGTH];
+    unsigned int iEncryptionSystem;
+    char strIconPath[PVR_ADDON_URL_STRING_LENGTH];
+    bool bIsHidden;
+    bool bHasArchive;
+    int iOrder;
+  } PVR_CHANNEL;
+
+  /*!
+   * @brief "C" PVR add-on signal status information
+   *
+   * Structure used to interface in "C" between Kodi and Addon.
+   *
+   * See @ref kodi::addon::PVRSignalStatus for description of values.
+   */
+  typedef struct PVR_SIGNAL_STATUS
+  {
+    char strAdapterName[PVR_ADDON_NAME_STRING_LENGTH];
+    char strAdapterStatus[PVR_ADDON_NAME_STRING_LENGTH];
+    char strServiceName[PVR_ADDON_NAME_STRING_LENGTH];
+    char strProviderName[PVR_ADDON_NAME_STRING_LENGTH];
+    char strMuxName[PVR_ADDON_NAME_STRING_LENGTH];
+    int iSNR;
+    int iSignal;
+    long iBER;
+    long iUNC;
+  } PVR_SIGNAL_STATUS;
+
+  //============================================================================
+  /// @ingroup cpp_kodi_addon_pvr_Defs_Channel_PVRDescrampleInfo
+  /// @brief Special @ref cpp_kodi_addon_pvr_Defs_Channel_PVRDescrampleInfo
+  /// value to indicate that a struct member's value is not available
+  ///
+  #define PVR_DESCRAMBLE_INFO_NOT_AVAILABLE -1
```

Comment:
In the old API, this was `const int PVR_DESCRAMBLE_INFO_NOT_AVAILABLE = -1;`. Why have you replaced type safe `const int` with type unsafe defines? - Here and at other places in the new API.

Reply:
See above, is related to "C", where this is the biggest waste way to have inside header.

Reply:
Should be no issue with proper multi include protection for PVR.h